[CORE-13279] - Support ACL sync for DR

michael-redpanda opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

This PR adds support for syncing ACLs from the source cluster to the destination cluster

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

* None

michael-redpanda committed:
cl: Introduce security_service

security_service is an abstract class that allows for access to the
security_frontend service to modify ACLs.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cl: Wire in security_service into manager

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cl/model: Added security settings sync config

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cl: Introduce security migrator task

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
k/s: Split out security module

Split out the security module so it can be used by other modules.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
k/s: Moved ACL entry to resource conversion

Moved parsing of ACL entries to the describe_acls_resource response to
the security module so it can be referenced by tests.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
k/c/test: Wire in describe_acls handler

Hold an `acl_store` in the cluster_mock and handle `describe_acls` API
messages

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
admin: Update convert to handle security settings

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
cl: Wire in security_migrator into service

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
dt: Enable SR for multi-cluster services

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
dt: Add test to check ACL migration

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda commented:
Force push:

* Fix failing tests

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/71781-->
<!--6ef18941-3c40-46b4-96f7-767093784c9c-->
<!--e914955f-88f2-4390-a073-479d0993a603-->
<!--20066f00-2a25-423c-b83c-f224d4410d66-->
<!--f93e1f94-f59d-48d5-9bd4-086db42a981b-->
<!--title-->
## Retry command for [Build#71781](https://buildkite.com/redpanda/redpanda/builds/71781)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkBasicTests.test_can_not_create_more_than_one_link
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkBasicTests.test_create_simple_link
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--88044b34-f890-45eb-b519-06fa7e12db44-->
<!--TEST RESULTS-->
<!--26912ae3-b198-4c67-a31a-11c19b366e2e-->
<!--TEST RESULTS-->
<!--9596558a-e1fc-4d49-a936-4d9e076e2c41-->
<!--TEST RESULTS-->
<!--c80fcd32-7653-42c1-8c48-045976f638b0-->
<!--TEST RESULTS-->
<!--6aef2adc-f4e8-4d5e-9390-30103779acc4-->
<!--TEST RESULTS-->
<!--ef2e092c-3d98-4271-b652-85e82d345275-->
<!--TEST RESULTS-->
<!--9f449808-c62c-4f49-bde1-502317cdb432-->
<!--TEST RESULTS-->
<!--f2b03092-5d23-455c-8c6d-9ad7286ce1ea-->
<!--TEST RESULTS-->
<!--e3f54b63-c731-4426-bf17-3cce9cf73e4c-->
<!--TEST RESULTS-->
<!--28f3f56e-5cf4-48d7-8037-c5022c1a3284-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71781">build#71781</a></summary>


| test_class           | test_method                            | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 |
|:---------------------|:---------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|
| ShadowLinkBasicTests | test_can_not_create_more_than_one_link | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71781#01991544-1e93-4afb-9ea5-76eb0f46747d | FAIL          | 0/21     | The test has failed across all retries |
| ShadowLinkBasicTests | test_can_not_create_more_than_one_link | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71781#01991547-8d05-4c91-a0d8-c0b53f6fe0ff | FAIL          | 0/21     | The test has failed across all retries |
| ShadowLinkBasicTests | test_create_simple_link                | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71781#01991544-1e94-4cec-9453-2212f7e5ab80 | FAIL          | 0/21     | The test has failed across all retries |
| ShadowLinkBasicTests | test_create_simple_link                | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71781#01991547-8d06-4217-bedd-5468f8d59c0d | FAIL          | 0/21     | The test has failed across all retries |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71794">build#71794</a></summary>


| test_class        | test_method                   | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:------------------|:------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| CrashReporterTest | test_redpanda_crash_reporting | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71794#01991661-9bb0-482c-b2c4-d1d285870a38 | FLAKY         | 20/21    | upstream reliability is '98.29268292682927'. current run reliability is '95.23809523809523'. drift is 3.05459 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71854">build#71854</a></summary>


| test_class                   | test_method        | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:-----------------------------|:-------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| PartitionMovementUpgradeTest | test_basic_upgrade | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71854#019929fb-000d-44b5-803a-6faeb68a1c9c | FLAKY         | 20/21    | upstream reliability is '97.82082324455206'. current run reliability is '95.23809523809523'. drift is 2.58273 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71931">build#71931</a></summary>


| test_class            | test_method         | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:----------------------|:--------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| PartitionBalancerTest | test_rack_awareness | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71931#01992fd2-b3c7-48c4-adab-9798079c8922 | FLAKY         | 20/21    | upstream reliability is '99.48717948717949'. current run reliability is '95.23809523809523'. drift is 4.24908 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71999">build#71999</a></summary>


| test_class                 | test_method                  | test_arguments                                      | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    |
|:---------------------------|:-----------------------------|:----------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|
| DatalakeClusterRestoreTest | test_slow_tiered_storage_dlq | {"catalog_type": "nessie", "cloud_storage_type": 1} | integration | https://buildkite.com/redpanda/redpanda/builds/71999#01993436-4af1-4301-a00b-5bd3c850df68 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72074">build#72074</a></summary>


| test_class         | test_method    | test_arguments            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    |
|:-------------------|:---------------|:--------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShardPlacementTest | test_node_join | {"disable_license": true} | integration | https://buildkite.com/redpanda/redpanda/builds/72074#0199390a-6c22-4f9c-b358-d520eb72ea43 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72164">build#72164</a></summary>


| test_class               | test_method                          | test_arguments                                                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:-------------------------|:-------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| AuditLogTestOauth        | test_kafka_oauth                     | {"authz_match": "rbac"}                                                                                                                        | integration | https://buildkite.com/redpanda/redpanda/builds/72164#01993ead-6fc3-455a-81d2-a10208551d1f | FLAKY         | 15/21    | upstream reliability is '91.10512129380054'. current run reliability is '71.42857142857143'. drift is 19.67655 and the allowed drift is set to 50. The test should PASS |
| ShadowLinkSecurityTests  | test_acl_sync                        | {"check_sr": false}                                                                                                                            | integration | https://buildkite.com/redpanda/redpanda/builds/72164#01993ead-6fbd-4ab2-a37b-44a55168b814 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| ShadowLinkSecurityTests  | test_acl_sync                        | {"check_sr": false}                                                                                                                            | integration | https://buildkite.com/redpanda/redpanda/builds/72164#01993eb5-8a87-44ba-bb6e-bdb9e7e204b2 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| ShadowLinkSecurityTests  | test_acl_sync                        | {"check_sr": true}                                                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/72164#01993ead-6fbf-4cd5-b553-83d7da86ebb7 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| ShadowLinkSecurityTests  | test_acl_sync                        | {"check_sr": true}                                                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/72164#01993eb5-8a88-4e9e-a143-d49db1a957b9 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  |
| DataMigrationsApiTest    | test_creating_and_listing_migrations | null                                                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/72164#01993eb5-8a8d-4f29-8a72-598bef115b91 | FLAKY         | 20/21    | upstream reliability is '99.14772727272727'. current run reliability is '95.23809523809523'. drift is 3.90963 and the allowed drift is set to 50. The test should PASS  |
| DataMigrationsApiTest    | test_higher_level_migration_api      | null                                                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/72164#01993ead-6fc0-4a13-8f58-c0b352c98327 | FLAKY         | 18/21    | upstream reliability is '99.65986394557824'. current run reliability is '85.71428571428571'. drift is 13.94558 and the allowed drift is set to 50. The test should PASS |
| RandomNodeOperationsTest | test_node_operations                 | {"cloud_storage_type": 1, "compaction_mode": "chunked_sliding_window", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/72164#01993eb5-8a89-4ed7-a2a3-3b3528217f47 | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS              |
| RapidTopicRecreateTest   | test_topic_rapid_recreation          | null                                                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/72164#01993ead-6fc7-4260-b48c-9c7e6af330e9 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72205">build#72205</a></summary>


| test_class                       | test_method                | test_arguments      | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                     |
|:---------------------------------|:---------------------------|:--------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkSecurityTests          | test_acl_sync              | {"check_sr": false} | integration | https://buildkite.com/redpanda/redpanda/builds/72205#01993fe8-cfb2-4be3-aa18-35d57c3e0cc1 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                     |
| ShadowLinkSecurityTests          | test_acl_sync              | {"check_sr": false} | integration | https://buildkite.com/redpanda/redpanda/builds/72205#01993feb-d6d5-4f2b-aa14-1b4d44ffd15b | FAIL          | 0/21     | The test has failed across all retries                                                                                                                     |
| ShadowLinkSecurityTests          | test_acl_sync              | {"check_sr": true}  | integration | https://buildkite.com/redpanda/redpanda/builds/72205#01993fe8-cfb3-4707-b169-f29d46118edc | FAIL          | 0/21     | The test has failed across all retries                                                                                                                     |
| ShadowLinkSecurityTests          | test_acl_sync              | {"check_sr": true}  | integration | https://buildkite.com/redpanda/redpanda/builds/72205#01993feb-d6d7-4807-8dbc-3f72c832c44e | FAIL          | 0/21     | The test has failed across all retries                                                                                                                     |
| WriteCachingFailureInjectionTest | test_unavoidable_data_loss | null                | integration | https://buildkite.com/redpanda/redpanda/builds/72205#01993feb-d6df-4579-8c9a-9a1a499b3e83 | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72590">build#72590</a></summary>


| test_class                | test_method                                                     | test_arguments                                                                                                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                                       |
|:--------------------------|:----------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EndToEndCloudTopicsTest   | test_write                                                      | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/72590#01996215-3d8b-42cc-8d9f-dea8a6290d25 | FLAKY         | 11/21    | upstream reliability is '98.17708333333334'. current run reliability is '52.38095238095239'. drift is 45.79613 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=EndToEndCloudTopicsTest&test_method=test_write                                                   |
| EndToEndCloudTopicsTxTest | test_write                                                      | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/72590#01996215-3d8c-44b2-9665-4038dd88713e | FLAKY         | 14/21    | upstream reliability is '93.69369369369369'. current run reliability is '66.66666666666666'. drift is 27.02703 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=EndToEndCloudTopicsTxTest&test_method=test_write                                                 |
| ClusterRateQuotaTest      | test_client_response_throttle_mechanism                         | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/72590#01996215-3d88-4075-acc2-38063e9bce50 | FLAKY         | 15/21    | upstream reliability is '97.51243781094527'. current run reliability is '71.42857142857143'. drift is 26.08387 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterRateQuotaTest&test_method=test_client_response_throttle_mechanism                         |
| ClusterRateQuotaTest      | test_client_response_throttle_mechanism_applies_to_next_request | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/72590#01996215-3d89-4319-a036-ebd04ba908ea | FLAKY         | 15/21    | upstream reliability is '100.0'. current run reliability is '71.42857142857143'. drift is 28.57143 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterRateQuotaTest&test_method=test_client_response_throttle_mechanism_applies_to_next_request |
| NodeWiseRecoveryTest      | test_recovery_local_data_missing                                | {"wait_for_final_manifest_uploads": false}                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/72590#0199620d-a269-4a22-9036-694da080fce2 | FLAKY         | 18/21    | upstream reliability is '100.0'. current run reliability is '85.71428571428571'. drift is 14.28571 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodeWiseRecoveryTest&test_method=test_recovery_local_data_missing                                |
| NodeWiseRecoveryTest      | test_recovery_local_data_missing                                | {"wait_for_final_manifest_uploads": false}                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/72590#01996215-3d8a-4f63-a491-f28551ac838f | FLAKY         | 14/21    | upstream reliability is '100.0'. current run reliability is '66.66666666666666'. drift is 33.33333 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodeWiseRecoveryTest&test_method=test_recovery_local_data_missing                                |
| NodeWiseRecoveryTest      | test_recovery_local_data_missing                                | {"wait_for_final_manifest_uploads": true}                                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/72590#0199620d-a26a-4fed-bad2-8118d570ab28 | FLAKY         | 12/21    | upstream reliability is '98.9769820971867'. current run reliability is '57.14285714285714'. drift is 41.83412 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodeWiseRecoveryTest&test_method=test_recovery_local_data_missing                                |
| NodeWiseRecoveryTest      | test_recovery_local_data_missing                                | {"wait_for_final_manifest_uploads": true}                                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/72590#01996215-3d8b-42cc-8d9f-dea8a6290d25 | FLAKY         | 7/21     | upstream reliability is '98.2587064676617'. current run reliability is '33.33333333333333'. drift is 64.92537 and the allowed drift is set to 50. The test should FAIL  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodeWiseRecoveryTest&test_method=test_recovery_local_data_missing                                |
| RandomNodeOperationsTest  | test_node_operations                                            | {"cloud_storage_type": 1, "compaction_mode": "adjacent_merge", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/72590#01996215-3d8c-44b2-9665-4038dd88713e | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RandomNodeOperationsTest&test_method=test_node_operations                                        |
| RecoveryModeTest          | test_rolling_restart                                            | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/72590#0199620d-a268-40ed-8787-b2ca80c841f3 | FLAKY         | 18/21    | upstream reliability is '100.0'. current run reliability is '85.71428571428571'. drift is 14.28571 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RecoveryModeTest&test_method=test_rolling_restart                                                |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72946">build#72946</a></summary>


| test_class                     | test_method                                       | test_arguments                                                                                                                             | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                         |
|:-------------------------------|:--------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingReplicationTests  | test_replication_basic                            | {"shuffle_leadership": true, "source_cluster_spec": {"cluster_type": "kafka", "kafka_quorum": "COMBINED_KRAFT", "kafka_version": "3.8.0"}} | integration | https://buildkite.com/redpanda/redpanda/builds/72946#019980e8-c807-47e5-9483-42c718011be9 | FLAKY         | 12/21    | upstream reliability is '97.03389830508475'. current run reliability is '57.14285714285714'. drift is 39.89104 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_basic                   |
| ClusterRateQuotaTest           | test_client_group_consume_rate_throttle_mechanism | null                                                                                                                                       | integration | https://buildkite.com/redpanda/redpanda/builds/72946#019980e8-c807-47e5-9483-42c718011be9 | FLAKY         | 17/21    | upstream reliability is '81.51898734177216'. current run reliability is '80.95238095238095'. drift is 0.56661 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterRateQuotaTest&test_method=test_client_group_consume_rate_throttle_mechanism |
| ClusterRateQuotaTest           | test_client_group_produce_rate_throttle_mechanism | null                                                                                                                                       | integration | https://buildkite.com/redpanda/redpanda/builds/72946#019980e8-c808-4076-b789-e65389d44c2f | FLAKY         | 10/21    | upstream reliability is '83.00518134715026'. current run reliability is '47.61904761904761'. drift is 35.38613 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterRateQuotaTest&test_method=test_client_group_produce_rate_throttle_mechanism |
| DataMigrationsApiTest          | test_higher_level_migration_api                   | null                                                                                                                                       | integration | https://buildkite.com/redpanda/redpanda/builds/72946#019980f0-d00a-48f5-8c8a-59e6d0b8e618 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_higher_level_migration_api                  |
| DatalakeCustomPartitioningTest | test_many_partitions                              | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1}                                                                                     | integration | https://buildkite.com/redpanda/redpanda/builds/72946#019980e8-c807-47e5-9483-42c718011be9 | FLAKY         | 12/21    | upstream reliability is '100.0'. current run reliability is '57.14285714285714'. drift is 42.85714 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DatalakeCustomPartitioningTest&test_method=test_many_partitions                    |
| TopicRecoveryTest              | test_prevent_recovery                             | {"cloud_storage_type": 2}                                                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/72946#019980f0-d00a-48f5-8c8a-59e6d0b8e618 | FLAKY         | 18/21    | upstream reliability is '97.54098360655738'. current run reliability is '85.71428571428571'. drift is 11.8267 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TopicRecoveryTest&test_method=test_prevent_recovery                                |
</details>
<!--/body-->

michael-redpanda commented:
Force push:

* Fixed issue with base test class

michael-redpanda commented:
Force push:

* Rebased off of dev to fix conflicts

bharathv commented on src/v/cluster_link/security_migrator.h:
```diff
@@ -0,0 +1,55 @@
+
+namespace cluster_link {
+
+class security_migrator : public controller_locked_task {
```
nit: class comment and some high level description of what it is syncing.

bharathv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,423 @@
+    } catch (const std::exception& e) {
+        auto msg = ssx::sformat("Failed to update metadata: {}", e.what());
+        vlog(logger().warn, "{}", msg);
+        std::ignore = change_state(model::task_state::link_unavailable, msg);
```
unrelated, I feel "link_error" is probably a more user friendly name IMO for temporary errors.

michael-redpanda replied:
I actually feel the opposite, that `link_error` conveys a more critical failure

bharathv replied:
I think "unavailable" gives an impression of a connectivity issue (which is fine) but seems to be used as a blanket error in all cases, even if (for example) there is a divergence in state and "sync" fails. Maybe we need more nuanced error codes? (or some hierarchy)

michael-redpanda replied:
Ah gotcha - yeah maybe my original design was lacking some of this nuance

bharathv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,423 @@
+    try {
+        co_await cluster.request_metadata_update();
+    } catch (const std::exception& e) {
+        auto msg = ssx::sformat("Failed to update metadata: {}", e.what());
+        vlog(logger().warn, "{}", msg);
```
nit: vlog(logger.warn, "Failed to....", e)?

michael-redpanda replied:
`msg` is passed to `change_state`

bharathv replied:
dohh!

bharathv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,423 @@
+
+    std::ranges::for_each(res, [this](cluster::errc ec) {
+        if (ec != cluster::errc::success) {
+            vlog(logger().warn, "Failure during ACL creation: {}", ec);
```
curious what would be the link state in this case? Technically this is a sync error indicating some divergence.

michael-redpanda replied:
I'm not sure we need to update the state if this fails, I guess it depends on the failure (which I admit I'm not digging into it here).  My feeling is that whatever error that does occur is temporary and will resolve the next time the task executes (e.g. maybe controller leadership is moving).

bharathv replied:
agreed, I was thinking how we'd bubble up these temporary errors .. not in this PR but generally ...we should probably expose some "per task error" and rollup to a link error state. 

```
rpk link status foo
acls -- error --  <time since last successful sync>..
CO -- good -- 
topic_foo ---

```

bharathv reviewed (commented):
lgtm .. seems mostly mechanical fwict, no controversial changes.

michael-redpanda commented:
Forc epush:

* Addressed some PR comments

michael-redpanda commented:
Force push:

* Reb ased off of dev to resolve conflicts

michael-redpanda commented:
Force push:

* Fixed merge conflict

michael-redpanda commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,423 @@
+        if (is_registry_resource(resource)) {
+            return kafka::details::to_kafka_resource_type(resource_type);
+        } else {
+            return kafka::details::to_kafka_registry_resource_type(
```
Good catch bot

michael-redpanda commented:
Force push:

* FIxed bug

michael-redpanda commented:
Force push:

* Rebased off of dev

michael-redpanda commented:
Force push:

* Rebased off of dev

mmaslankaprv commented on src/v/cluster_link/model/types.h:
```diff
@@ -370,6 +370,190 @@ struct consumer_groups_mirroring_config
+    group,
+    topic,
+    txn_id,
+    sr_subject,
```
nit: i know it is more verbose but maybe we can use full name i.e. `schema_registry_subject` ?

mmaslankaprv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,423 @@
+    // Perform a metadata update to get as fresh as possible data from the
+    // source cluster
+    try {
+        co_await cluster.request_metadata_update();
```
do we need this update to happen every time here ?, if so maybe we should request an update without topics included ?

michael-redpanda replied:
Yeah I just need to ensure we have the appropriate cluster level authorizations to do describe ACLs

michael-redpanda replied:
I made an update to the `request_metadata_update` method, would appreicate your feedback on my approach, thank you

mmaslankaprv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,423 @@
+        vlog(logger().warn, "Error transforming received ACLs: {}", e.what());
+        co_return;
+    }
+    vlog(logger().trace, "bindings: {}", bindings);
```
nit: add some context to this log line

mmaslankaprv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,423 @@
+    vlog(logger().trace, "bindings: {}", bindings);
+
+    auto res = co_await get_link()->get_security_service().create_acls(
+      std::move(bindings), 5s);
```
define a timeout constant ? maybe this should be configurable ?

mmaslankaprv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,423 @@
+    }
+    vlog(logger().trace, "bindings: {}", bindings);
+
+    auto res = co_await get_link()->get_security_service().create_acls(
```
Do we need to support ACL deletion ?

michael-redpanda replied:
No, I guess how could we tell the difference between a user changing their filters (so we don't see the ACL when we describe them) vs them being deleted on the source.

mmaslankaprv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,423 @@
+
+    chunked_vector<kafka::describe_acls_resource> acls;
+
+    for (auto& req : requests) {
```
max concurrent for each ?

michael-redpanda replied:
Just to be clear, you want to send out the the describe_acls_request concurrently?

mmaslankaprv replied:
yeah, exactly

mmaslankaprv replied:
they are for different brokers

michael-redpanda replied:
yeah good point

bharathv reviewed (dismissed):
don't have more comments, I'll Michal sign off.

michael-redpanda commented:
Force push:

* Rebased off of dev

michael-redpanda commented:
Force push - addressed PR comments:

* spelled out `schema_registry` rather than `sr`
* Added ability to select which type of metadata update to perform: no topics, all topics, or a subset
* Now describe ACLs from source cluster concurrently

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/72164-->
<!--59496fd0-1c3a-417e-a37d-df07cabe9711-->
<!--cb426276-b1d4-466b-891f-1f4d1a235e75-->
<!--061d7ef8-6032-4366-8b93-585995a0c572-->
<!--8cb0ec90-bd46-4c52-a4b4-2b162a653116-->
<!--title-->
## Retry command for [Build#72164](https://buildkite.com/redpanda/redpanda/builds/72164)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkSecurityTests.test_acl_sync@{"check_sr":true}
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkSecurityTests.test_acl_sync@{"check_sr":false}
```
<!--/body-->

michael-redpanda commented:
Force push:

* ruff

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/72205-->
<!--7357344f-fbd9-4a47-b2e5-579bb3237309-->
<!--94d00ec6-8c60-405e-8cf7-74e2a3fdd113-->
<!--630ec7b0-e58a-4e8f-88e8-6e176bac7359-->
<!--c495fe8a-7804-498c-af7d-b48ee57fc6e6-->
<!--title-->
## Retry command for [Build#72205](https://buildkite.com/redpanda/redpanda/builds/72205)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkSecurityTests.test_acl_sync@{"check_sr":true}
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkSecurityTests.test_acl_sync@{"check_sr":false}
```
<!--/body-->

michael-redpanda commented:
Something weird happening here - reproducing it locally, shadow broker thinks it's sending a request with `include_cluster_authorized_operations` set to true, but source broker is seeing it as false... so not getting cluster level authorized ops so ACL syncing is not happening.

michael-redpanda commented:
Ahh I know what the problem is.... metadata v12 doesn't support cluster level authorized operations anymore, that got dropped in v10 for the describecluster API

michael-redpanda commented:
Need to pause this to update `kafka::cluster` to support the `DescribeClusterAPI`

michael-redpanda commented:
Need #27557 to merge to get DescribeCluster API support added

michael-redpanda commented:
Force push:

* Rebased off of dev and addressed some PR comments

michael-redpanda commented:
Force push:

* Some changes suggested by the bot
* Fixed python type lint error

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/72590-->
<!--b77b3f69-8b45-42cc-8c4f-45ec87481977-->
<!--title-->
## Retry command for [Build#72590](https://buildkite.com/redpanda/redpanda/builds/72590)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/partition_force_reconfiguration_test.py::NodeWiseRecoveryTest.test_recovery_local_data_missing@{"wait_for_final_manifest_uploads":true}
```
<!--/body-->

michael-redpanda commented:
Force push:

* Rebased off of dev to fix merge conflicts

mmaslankaprv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,429 @@
+    auto res = co_await get_link()->get_security_service().create_acls(
+      std::move(bindings), acl_creation_timeout);
+
+    std::ranges::for_each(res, [this](cluster::errc ec) {
+        if (ec != cluster::errc::success) {
+            vlog(logger().warn, "Failure during ACL creation: {}", ec);
```
I think we should check here if an error is retryable or not, also if everything failed we should still indicate it as an error. Logging error with warn is equal to just ignoring it

michael-redpanda replied:
> I think we should check here if an error is retryable or not,

We will technically retry in the sense that the task runs periodically.  Also do we have a list of retryable `cluster::errc`'s?    Looking in `src/v/cluster` and not seeing any method but maybe it's somewhere else?

> still indicate it as an error. Logging error with warn is equal to just ignoring it

Maybe transition the task state to some sort of faulted but recoverable state?

mmaslankaprv commented on src/v/cluster_link/security_migrator.cc:
```diff
@@ -0,0 +1,429 @@
+    auto acls = co_await fetch_acls(describe_acls_version);
+    vlog(logger().trace, "Fetched ACLs: {}", acls);
+
+    std::vector<security::acl_binding> bindings;
```
why not chunked vector ?

michael-redpanda replied:
`security_frontend::create_acls` takes a `std::vector`.  I could update the API to take a chunked vector but didn't want the review to get too much bigger.  But if you feel strongly we should change the container happy to do so.

mmaslankaprv commented:
just two small comments, otherwise looks good
